pkg/controllers/provisioningv2/cluster/controller.go

```diff
@@ -300,6 +300,21 @@ func (h *handler) createNewCluster(cluster *v1.Cluster, status v1.ClusterStatus,
 		})
 	}
 
+	if cluster.Spec.ClusterAgentDeploymentCustomization != nil {
```

Comment:
I see this handles updating creation of `clusters.management.cattle.io/v3` objects based on `clusters.provisioning.cattle.io/v1` objects.
Apologies if I've missed it, but where are we handling updating sync the other way around (creation of `clusters.provisioning.cattle.io/v1` based on `clusters.management.cattle.io/v3`)?